### PR TITLE
Implement Collect

### PR DESCRIPTION
**Pull request by mmatera**

### CODE REVIEW

**Comment by rocky** [comment]
Now that I know to look for it. Are there any other Attributes that `Collect` has in addition to being protected?  (I checked that it is neither Listable nor NumericFunction. But there are lots of other properties out there.

File: mathics/builtin/algebra.py [hunk 1]
```diff
@@ -1413,3 +1413,202 @@ def apply(self, expr, form, h, evaluation):
...
+
+            return Expression("Plus", *terms)
+        else:
+            if filt:
+                return Expression(filt, expr).evaluate(evaluation)
+            else:
+                return expr
+
+
+# tejimeto
```
Comment (rocky) [inline mathics/builtin/algebra.py:1614]: Google translate doesn't recognize this as a Spanish word.

File: CHANGES.rst [hunk 2]
```diff
@@ -14,7 +14,7 @@ New builtins
 * ``Series``,  ``O`` and ``SeriesData``
 * ``StringReverse``
 * Add all of the named colors, e.g. ``Brown`` or ``LighterMagenta``.
-
+* ``Collect``
```
Comment (rocky) [inline CHANGES.rst:17]: (I've been keeping these in alphabetic order. Not a big deal though.)

**Comment by mmatera** [comment]
> Now that I know to look for it. Are there any other Attributes that `Collect` has in addition to being protected? (I checked that it is neither Listable nor NumericFunction. But there are lots of other properties out there.
In WMA, 
```
In[1]:= Attributes[Collect]
Out[1]= {Protected}
```

File: mathics/builtin/algebra.py (hunk 1 above)
Comment (mmatera) [reply mathics/builtin/algebra.py:1614]: ja! Sorry, that was a personal note, about a  word my daughter invented...

File: CHANGES.rst (hunk 2 above)
Comment (mmatera) [reply CHANGES.rst:17]: OK, when I finish cleaning this, I will put in the right place..

**Comment by mmatera** [comment]
Superseeded by  #1324

**Pull request closed without merging**